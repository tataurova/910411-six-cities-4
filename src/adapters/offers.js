export const getOffer = (data) => {
  return {
    id: data.id,
    title: data.title,
    city: data.city.name,
    cityCoordinates: [data.city.location.latitude, data.city.location.longitude],
    cityZoom: data.city.location.zoom,
    coordinates: [data.location.latitude, data.location.longitude],
    type: data.type,
    price: data.price,
    rating: data.rating,
    premium: data.is_premium,
    favorite: data.is_favorite,
    photo: data.preview_image,
    bedrooms: data.bedrooms,
    adults: data.max_adults,
    additional: data.goods,
    details: data.description,
    owner: {
      photo: data.host.avatar_url,
      name: data.host.name,
      isSuper: data.host.is_pro,
    },
    photos: data.images,
    reviews: [ // todo reviews from server
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  };
};
