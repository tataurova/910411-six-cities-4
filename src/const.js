export const PLACES_COUNT = 312;

export const MapSettings = {
  city: [52.38333, 4.9],
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30],
  },
  zoom: 12,
};
