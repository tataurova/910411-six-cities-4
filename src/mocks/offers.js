export default [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.960361, 6.967974],
    type: `Apartment`,
    price: 120,
    rating: 4,
    premium: true,
    photo: `img/apartment-01.jpg`,
    bedrooms: 3,
    adults: 6,
    additional: [`Wifi`, `Heating`, `Kitchen`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: true,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.913361, 6.9509739999999995],
    type: `Private room`,
    price: 80,
    rating: 4,
    premium: false,
    photo: `img/room.jpg`,
    bedrooms: 2,
    adults: 4,
    additional: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-max.jpg`,
      name: `Max`,
      isSuper: false,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];

/* eslint-disable */
export const serverOffers = [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.960361,
      longitude: 6.967974,
    },
    type: `Apartment`,
    price: 120,
    rating: 4,
    is_premium: true,
    preview_image: `img/apartment-01.jpg`,
    bedrooms: 3,
    max_adults: 6,
    goods: [`Wifi`, `Heating`, `Kitchen`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 1,
      avatar_url: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      is_pro: true,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.913361,
      longitude: 6.9509739999999995,
    },
    type: `Private room`,
    price: 80,
    rating: 4,
    is_premium: false,
    preview_image: `img/room.jpg`,
    bedrooms: 2,
    max_adults: 4,
    goods: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 2,
      avatar_url: `img/avatar-max.jpg`,
      name: `Max`,
      is_pro: false,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];
