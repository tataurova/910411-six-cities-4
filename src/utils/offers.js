export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  return {
    offer: offers.filter((item) => item.id === idNumber)[0],
    nearOffers: offers.filter((item) => item.id !== idNumber),
  };
  // for (let key in offers) {
  //   if (offers.hasOwnProperty(key)) {
  //     const offer = offers[key].filter((item) => item.id === idNumber)[0];
  //     if (offer) {
  //       return {
  //         offer,
  //         nearOffers: offers[key].filter((item) => item.id !== offer.id),
  //       };
  //     }
  //   }
  // }
  // return {};
};
