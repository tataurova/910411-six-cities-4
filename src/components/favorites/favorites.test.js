import React from "react";
import renderer from "react-test-renderer";
import Favorites from "./favorites.tsx";
import {BrowserRouter} from "react-router-dom";
import offers from "../../mocks/offers.js";
import configureStore from "redux-mock-store";
import {CardType} from "../../const.js";
import {AuthorizationStatus} from "../../const.js";
import {DEFAULT_ERROR_STATUS} from "../../const";

const onBookmarkButtonCLick = jest.fn();
const mockStore = configureStore([]);
const initialState = {
  onBookmarkButtonCLick,
};
const store = mockStore(initialState);

describe(`<Favorites />`, () => {
  it(`Render Favorites`, () => {
    const testUser = `test@test.ru`;
    const tree = renderer
      .create(
          <BrowserRouter>
            <Favorites
              store = {store}
              cardType = {CardType.FAVORITE}
              authorizationStatus = {AuthorizationStatus.AUTH}
              user = {testUser}
              favoriteOffers={offers}
              loadFavoriteOffers={() => {}}
              error = {DEFAULT_ERROR_STATUS}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
