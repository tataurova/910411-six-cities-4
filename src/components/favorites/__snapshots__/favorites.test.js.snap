// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> Render Favorites 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  test@test.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <div
            className="favorites__places"
          >
            <li
              className="favorites__locations-items"
            >
              <div
                className="favorites__locations locations locations--current"
              >
                <div
                  className="locations__item"
                >
                  <a
                    className="locations__item-link"
                    href="#"
                  >
                    <span>
                      Cologne
                    </span>
                  </a>
                </div>
              </div>
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="img/apartment-01.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info "
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        120
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button "
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </li>
            <li
              className="favorites__locations-items"
            >
              <div
                className="favorites__locations locations locations--current"
              >
                <div
                  className="locations__item"
                >
                  <a
                    className="locations__item-link"
                    href="#"
                  >
                    <span>
                      Cologne
                    </span>
                  </a>
                </div>
              </div>
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="img/room.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info "
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        80
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button "
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/2"
                      onClick={[Function]}
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private room
                  </p>
                </div>
              </article>
            </li>
          </div>
        </ul>
      </section>
    </div>
  </main>,
]
`;
